bson/src/main/org/bson/codecs/pojo/PojoCodecProvider.java

```diff
@@ -72,6 +74,17 @@ public static Builder builder() {
                 classModel = createClassModel(clazz, conventions);
             }
             return new PojoCodecImpl<T>(classModel, registry, discriminatorLookup);
+        } else if (automatic) {
+            try {
+                classModel = createClassModel(clazz, conventions);
+            } catch (IllegalStateException e) {
+                return null;
+            }
+            if (!clazz.isInterface() && classModel.getPropertyModels().isEmpty()) {
+                return null;
+            }
+            discriminatorLookup.addClassModel(classModel);
```

Comment:
I'm not clear why discriminator lookup is handled differently for automatic pojo creation and lazy creation based on a package name match?  Could the two paths be unified?